sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CustomCoder.java

```diff
@@ -64,16 +64,18 @@
   }
 
   /**
-   * Returns an empty list. A {@link CustomCoder} may not have argument coders
+   * {@inheritDoc}.
+   *
+   * <p>Returns an empty list. A {@link CustomCoder} has no default argument {@link Coder coders}.
    */
   @Override
-  public final List<Coder<?>> getCoderArguments() {
-    return null;
+  public List<Coder<?>> getCoderArguments() {
+    return Collections.emptyList();
   }
 
   /**
-   * Returns an empty list. A {@link CustomCoder} may not have component coders that are used for
-   * inference.
+   * Returns an empty list. A {@link CustomCoder} by default will not have component coders that are
+   * used for inference.
    */
   public static <T> List<Object> getInstanceComponents(T exampleValue) {
```

Comment:
Do we need this method?

Since its static, I would expect that subclasses will need to define their own static versions of this.